autosklearn/automl.py

```diff
@@ -1834,15 +1834,52 @@ def get_models_with_weights(self):
         return self.ensemble_.get_models_with_weights(self.models_)
 
     def show_models(self):
-        models_with_weights = self.get_models_with_weights()
-
-        with io.StringIO() as sio:
-            sio.write("[")
-            for weight, model in models_with_weights:
-                sio.write("(%f, %s),\n" % (weight, model))
-            sio.write("]")
-
-            return sio.getvalue()
+        """ Returns a dictionary containing models and their information
+            where model_id/run_id is the key """
```

Comment:
A feature is only discovered through documentation and examples so we need some documentation for it. You can see how it's rendered [code](https://github.com/automl/auto-sklearn/blob/master/autosklearn/estimators.py#L557) / [doc](https://automl.github.io/auto-sklearn/master/api.html#autosklearn.classification.AutoSklearnClassifier.leaderboard).

If you want people to use this feature (which we do), it needs to have friendly documentation. Perhaps you could give an example output as you do.

Some things to mention:
* Only returns models in the ensemble.
* What is in the dictionary returned by this method
* Give a small example output (just replace anything too long with `...`)